src/test/java/reactor/kafka/sender/internals/MockSenderTest.java

```diff
@@ -421,7 +420,6 @@ public void sendWithResponseStopOnError() {
                     .recordWith(() -> sendResponses)
                     .expectError(InvalidTopicException.class)
                     .verify(Duration.ofMillis(DEFAULT_TEST_TIMEOUT));
-        assertEquals(maxInflight, outgoing.onNextCount.get());
```

Comment:
the assertion wasn't correct (breaks due to prefetching). We have `producer.enableInFlightCheck();` a few lines earlier that will ensure that `maxInflight` is correct.